core/src/main/java/me/mykindos/betterpvp/core/logging/UuidLogger.java

```diff
@@ -0,0 +1,218 @@
+     */
+    public static void AddItemUUIDMetaInfo(@NotNull UUID logUUID, @NotNull UUID itemUuid, @NotNull UuidLogType type, @Nullable UUID uuid, @NotNull UUIDType uuidType) {
+        UUID metaUUID = UUID.randomUUID();
+
+
+        String query = "INSERT INTO uuidlogmeta (id, LogUUID, ItemUUID, Type, UUID, UUIDtype) VALUES (?, ?, ?, ?, ?, ?)";
+        database.executeUpdate(new Statement(query,
+                new UuidStatementValue(metaUUID),
+                new UuidStatementValue(logUUID),
+                new UuidStatementValue(itemUuid),
+                new StringStatementValue(type.name()),
+                new StringStatementValue(uuid == null ? null : uuid.toString()),
+                new StringStatementValue(uuidType.name())
+                )
+        );
+    }
+
+    /**
+     *
+     * @param itemUUID the uuid of the item
+     * @param amount the number of logs to retrieve
+     * @return A list of the last amount of logs relating to this uiid
+     */
+    public static List<String> getUuidLogs(UUID itemUUID, int amount) {
+        List<String> logList = new ArrayList<>();
+        if (amount < 0) {
+            return logList;
+        }
+
+        String query = "CALL GetUuidLogsByUuid(?, ?)";
+        CachedRowSet result = database.executeQuery( new Statement(query,
+                new UuidStatementValue(itemUUID),
+                new IntegerStatementValue(amount)
+                )
+        );
+
+        try {
+            while (result.next()) {
+                long time = result.getLong(1);
+                logList.add("<green>" + UtilTime.getTime(System.currentTimeMillis() - time, 2) + " ago</green> " + result.getString(2));
+            }
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        return logList;
+    }
+
+    /**
+     *
+     * @param playerUuid the uuid of the player
+     * @param amount the number of logs to retrieve
+     * @return A list of the last amount of logs relating to this player
+     */
+    public static List<String> getPlayerLogs(UUID playerUuid, int amount) {
+        List<String> logList = new ArrayList<>();
+        if (amount < 0) {
+            return logList;
+        }
+
+        String query = "CALL GetUuidLogsByPlayer(?, ?)";
+        CachedRowSet result = database.executeQuery( new Statement(query,
+                        new UuidStatementValue(playerUuid),
+                        new IntegerStatementValue(amount)
+                )
+        );
+
+        try {
+            while(result.next()) {
+                long time = result.getLong(1);
+                logList.add("<green>" + UtilTime.getTime(System.currentTimeMillis() - time, 2) + " ago</green>  " + result.getString(2));
+            }
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        return logList;
+    }
+
+
+    public static UUID legend(String message, Object... args) {
+        return log("LEGEND", message, args);
```

Comment:
LEGEND isnt a log level, log level is a severity rating, so typically they fall into one of these

FATAL, ERROR, WARN, INFO, DEBUG, and TRACE